### PR TITLE
Documentation around error propagation & version bumps

### PR DESCRIPTION
**Pull request by twokul**

### CODE REVIEW

File: ERRORS.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  + `location`
+  + `message`
+  + `stack`
+
+Error contains information of where in the file error occured and what kind of error it was.
+
++ `ember-cli-htmlbars` will catch the error from `ember-template-compiler`.
+
+Because `ember-cli-htmlbars` is based on `broccoli-persistent-filter`, technically `broccoli-persistent-filter`
+will catch the exception first. For example, we attach `file` and `treeDir` information to errors [here](https://github.com/stefanpenner/broccoli-persistent-filter/blob/master/index.js#L283-L288).
```
Comment (rwjblue) [inline ERRORS.md]: Link should be to a tag (so I doesn't drift and point to the wrong code as changes are made upstream)...

File: ERRORS.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+Note, that is just an interface, not an actual class.
+
+```typescript
+interface Position {
+  line: number;
+  column: number;
+}
+
+interface BuildError {
+  stack: string;
```
Comment (rwjblue) [inline ERRORS.md]: Maybe add some JSDOC style comments for each property with a bit more details? I'm not sure each peoperty needs fleshing out, but some do (e.g. "What the heck is a 'code frame'?").

**Comment by stefanpenner** [comment]
lets put this in the `doc/` folder

File: ERRORS.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+ * @param {string} location - The position of the error in the file.
+ */
+interface BuildError {
+  stack: string;
+  message: string;
+  codeFrame: string;
+  type: string,
+  location: Position;
+}
+```
```
Comment (stefanpenner) [inline ERRORS.md]: might be worth adding a quick real example towards the end. Of how doing the "right thing"

**Comment by stefanpenner** [comment]
@twokul is this ready to merge?

**Comment by twokul** [comment]
@stefanpenner addressed all of your comments, I believe it is 📦 :shipit: 🚀

**Comment by stefanpenner** [comment]
@twokul sweet. Once tests are green, lets tell homu to do its thing.

**Comment by stefanpenner** [comment]
@homu r+

**Pull request merged**